bing-wallpaper.sh

```diff
@@ -98,7 +103,18 @@ read -ra urls < <(curl -sL $PROTO://www.bing.com | \
     grep -Eo "url:'.*?'" | \
     sed -e "s/url:'\([^']*\)'.*/$PROTO:\/\/bing.com\1/" | \
     sed -e "s/\\\//g" | \
-    sed -e "s/\([[:digit:]]*x[[:digit:]]*\)/$RESOLUTION/")
+    sed -e "s/\([[:digit:]]*x[[:digit:]]*\)/$RESOLUTION/" | \
+    tr "\n" " ")
+
+if [ $BOOST ]; then
+    read -ra archiveUrls < <(curl -sL "$PROTO://www.bing.com/HPImageArchive.aspx?format=js&n=$BOOST" | \
```

Comment:
It looks like this is duplicating the regexps from above. Is there any way we could extract this functionality out into a function that just takes a URL?